src/server/cmd/job-shim/main.go

```diff
@@ -47,6 +47,23 @@ func do(appEnvObj interface{}) error {
 				os.Exit(0)
 			}
 
+			// Make sure that we call FinishJob even if something caused a panic
+			defer func() {
+				if r := recover(); r != nil {
+					if _, err := ppsClient.FinishJob(
```

Comment:
In general I think it's good to defend against things like this even when it seems hard for the current code to exhibit the bug. Someone could easily come along and add some buggy code after the call the `FinishJob` and wind up with an incredibly confusing error to debug.

But also I disagree that there isn't room for panicing because we `defer` another function on line `85` to unmount the volume. If that function panics then we'll hit this bug. So all it really takes is someone making a mistake in `Unmount` and we'll hit this issue.

I think it's worth an extra boolean variable to defend against that case.

Also could we add a log statement when we recover so we know that it's a bug in the job-shim that caused the failure rather than an actually failure? I think that'd help with debugging too.